core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -916,11 +945,9 @@ private[spark] class BlockManager(
     }
 
     if (blockWasSuccessfullyStored) {
-      DoPutSucceeded
-    } else if (iteratorFromFailedMemoryStorePut.isDefined) {
-      DoPutIteratorFailed(iteratorFromFailedMemoryStorePut.get)
+      None
     } else {
-      DoPutBytesFailed
+      Some(iteratorFromFailedMemoryStorePut.get)
```

Comment:
I would just return `iteratorFromFailedMemoryStorePut` here directly. Doing the `Some(opt.get)` is a strange pattern. If you want you can also add an `assert(blockWasSuccessfullyStored == iteratorFromFailedMemoryStorePut.isEmpty)`.